http-client/Network/HTTP/Client/Manager.hs

```diff
@@ -205,7 +205,13 @@ getConn req m
     -- Stop Mac OS X from getting high:
     -- https://github.com/snoyberg/http-client/issues/40#issuecomment-39117909
     | S8.null h = throwHttp $ InvalidDestinationHost h
-    | otherwise = takeKeyedPool (mConns m) connkey
+    | otherwise = 
+        -- Release connection in case of connection timeout:
+        -- https://github.com/snoyberg/http-client/pull/454
+        bracketOnError
```

Comment:
This call is a nop. `bracketOnError` will only trigger the cleanup if the in-between action throws an exception. If the allocation action throws an exception, the finalizer will never be run.

Reply:
I thought about that. On the other hand, in case of exception `Managed Connection` will not be created and it will not be an issue. Is it correct or I am missing something?

Reply:
That's correct. I'm not sure what problem you're trying to prevent here.

Reply:
- I was concerned about `timeout timeout' (getConn req m)` call in `getConnectionWrapper`. 
- I thought that in some tricky case connection could be acquired but dropped via `timeout`. In this case its future will be uncertain and I tried to prevent it.
- If `Managed Connection` created before timeout, it definitely would be returned. Otherwise, it would not be created at all. 
- And if it would not be created, then `ConnectionTimeout` exception could be "safely" thrown from `getConnectionWrapper`.

Now I see, there is nothing to worry about. I will completely restore `getConn`. 

Thank you for your comment.

Reply:
Fixed.